Ponca/src/SpatialPartitioning/KnnGraph/Iterator/knnGraphRangeIterator.h

```diff
@@ -11,31 +11,71 @@ namespace Ponca {
 template <typename Traits>
 class KnnGraphRangeQuery;
 
+/*!
+ *  \brief Input iterator to read the `KnnGraphRangeQuery`.
+ *
+ *  As this is an input iterator, we don't guarantee anything else than reading and incrementing values with it.
+ *  If you need to analyse the values with algorithms that relies on forward or more complex iterators,
+ *  we suggest copying the values inside a std::vector<Index>.
+ *
+ *  To restart the `KnnGraphRangeIterator`, for more than one iteration pass,
+ *  the KnnGraph queue must be reset using `KnnGraphRangeQuery::initialize(Iterator& iterator)`.
```

Comment:
I found this part of the documentation very confusing. I think we should not allow that, and we always ask the query for a new iterator if we need to restart the evaluation. Iterators are not meant to be stored.